src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
It's not blocking me at all as I already maintain a fork of this package for my project (OpenBazaar in the browser), so i will use my simple async.each solution in my fork, and leave it up to you to make a better fix whenever you have time. I would argue it's worth using async.each here in the meantime, as the current implementation is fairly unusable unless you manually filter addresses to be valid, but that's totally up to you guys.